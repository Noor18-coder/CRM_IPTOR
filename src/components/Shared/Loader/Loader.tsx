import { useMediaQuery } from 'react-responsive';

export interface Props {
  component?: string;
}

const Loader: React.FC<Props> = ({ component }) => {
  const isMobile = useMediaQuery({ maxWidth: 767 });
  const isTablet = useMediaQuery({ minWidth: 768, maxWidth: 991 });
  return (
    <div id="loader-overlay">
      <div id={component ? 'loader_container_opp' : 'loader_container'}>
        <div id="loader-bg">
          <div id="loader_part_1_cover">
            <div id="loader_part_1" />
          </div>
          <div id="loader_part_2_cover">
            <div id="loader_part_2" />
          </div>
          <div id="loader_part_3_cover">
            <div id="loader_part_3" />
          </div>
          <span id="loader-cover" />
        </div>
        <p id={component || isMobile || isTablet ? 'loader-text_opp' : 'loader-text'}>Loading...</p>
      </div>
    </div>
  );
};

export default Loader;
