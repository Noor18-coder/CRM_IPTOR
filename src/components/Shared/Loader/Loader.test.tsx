import renderer from 'react-test-renderer';
import Loader from './Loader';

describe('[Shared] Loader', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
