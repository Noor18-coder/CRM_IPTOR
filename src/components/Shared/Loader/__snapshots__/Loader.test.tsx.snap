// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Shared] Loader should renders correctly 1`] = `
<div
  id="loader-overlay"
>
  <div
    id="loader_container"
  >
    <div
      id="loader-bg"
    >
      <div
        id="loader_part_1_cover"
      >
        <div
          id="loader_part_1"
        />
      </div>
      <div
        id="loader_part_2_cover"
      >
        <div
          id="loader_part_2"
        />
      </div>
      <div
        id="loader_part_3_cover"
      >
        <div
          id="loader_part_3"
        />
      </div>
      <span
        id="loader-cover"
      />
    </div>
    <p
      id="loader-text"
    >
      Loading...
    </p>
  </div>
</div>
`;
